BLM updates, WAR IR checks

lokibunny opened the pull request:
BLM - working on manafont timing - still in progress
WAR - added level checks to help with sync issues

lokibunny committed:
Small update

Fixed redraw for AST cards
Adjusted priority for stack burn for foul
Added time in combat check for BLM buffs
WHM finally has multi DoT

lokibunny committed:
RDM changes

Allow Embolden to just be cast on CD
Added toggle for Magick Barrier

lokibunny committed:
RDM Update

Added checks for AoE abilities so they don't interrupt melee combo

lokibunny committed:
BLM fix

Fixed umbral soul spam

lokibunny committed:
Merge branch 'master' into master

lokibunny committed:
BLM update

Added check for aura for sharpcast so its not doublecast

lokibunny committed:
Merge branch 'master' of https://github.com/lokibunny/MagitekRoutine

lokibunny committed:
RDM update

Fixed logic causing verthunder and veraero to drop through and hardcast without dualcast up

lokibunny committed:
RDM update

Added weaving logic to oGCDs to alleviate spam
Modified logic for detecting combo status to be better with level synced content

lokibunny committed:
Merge branch 'master' into master

lokibunny committed:
BLM updates, WAR IR checks

BLM - working on manafont timing - still in progress
WAR - added level checks to help with sync issues

lokibunny committed:
Merge branch 'master' of https://github.com/lokibunny/MagitekRoutine

lokibunny committed:
Merge branch 'master' into master

lokibunny committed:
WAR onslaught

Modified check to make sense for saving charges

lokibunny committed:
Merge branch 'master' of https://github.com/lokibunny/MagitekRoutine

wildchill commented on Magitek/Logic/Warrior/Buff.cs:
```diff
@@ -41,10 +41,13 @@ public static async Task<bool> InnerRelease()
                 return false;
 
-            if (Combat.Enemies.Count(r => r.Distance(Core.Me) <= 3 + r.CombatReach) < 1)
+            //Radius of WAR AoEs is 5y not 3y, adding combat reach should be okay
```
I assume this is 3y because it's checking for an enemy being in range for fell cleave, not for AOE. I don't think this is a bad change necessarily since the player is probably going to move into range, but if for some reason they can't then this would cause a wasted IR.

wildchill commented on Magitek/Logic/Warrior/SingleTarget.cs:
```diff
@@ -157,7 +157,8 @@ public static async Task<bool> Onslaught()
             }
 
-            if (Spells.Onslaught.Charges < WarriorSettings.Instance.SaveOnslaughtCharges + 1)
+            //Why + 1 just keep the number of charges specified - also if you want to keep that number it should be <= not just <
```
I think this is because Spells.Onslaught.Charges is a float and it increases fractionally as a charge goes through its cooldown.

With your change if someone has SaveOnslaughtCharges set to 1, as soon as Charges goes to 1.1 then a charge would be used, which would result in the player not having a charge available any more despite a setting that indicated they wanted to save 1 charge.

wildchill reviewed (changes requested):
Check out my feedback and let me know what you think, thanks!

wildchill commented on Magitek/Logic/BlackMage/SingleTarget.cs:
```diff
@@ -315,10 +323,15 @@ public static async Task<bool> Blizzard3()
             //Post 72 logic
             if (Core.Me.ClassLevel > 71)
             {
+                if (await Spells.ManaFont.Cast(Core.Me))
```
Seems like this is duplicated, inside the ClassLevel check and after it. Could place it once above the if statement?

lokibunny committed:
WAR changes adjusted

Informed of logic for charges and fell cleave

lokibunny reviewed (commented):
Looks good to me

lokibunny committed:
BLM update

consolidated manafont check in B3

lokibunny committed:
BLM update

Still working manafont - added check for xeno cast with 0 mp.

lokibunny committed:
RDM and BL:M updates

RDM routine should now work as expected at lvl 90

BLM - fixed manafont - triple and quad flare now work relatively reliably.

wildchill commented on Magitek/Logic/Warrior/SingleTarget.cs:
```diff
@@ -157,7 +157,7 @@ public static async Task<bool> Onslaught()
             }
 
-            if (Spells.Onslaught.Charges < WarriorSettings.Instance.SaveOnslaughtCharges + 1)
+            if (Spells.Onslaught.Charges <= WarriorSettings.Instance.SaveOnslaughtCharges + 1)
```
I don't think you want `<=` here. Remember that `Spells.Onslaught.Charges` is a float value.
Assume:
- A person wants to save 2 charges
- There are 3 charges max

`Spells.Onslaught.Charges` will climb to 3.0 since there are 3 charges max for the job, the player will have set 2 into `WarriorSettings.Instance.SaveOnslaughtCharges` because they want to save 2 charges.

This condition will continue to return false because 3 <= 2 + 1 is true and will return false.

wildchill commented on Magitek/Logic/Summoner/Aoe.cs:
```diff
@@ -49,7 +49,7 @@ public static async Task<bool> Deathflare()
             var target = Combat.SmartAoeTarget(Spells.Deathflare, SummonerSettings.Instance.SmartAoe);
 
-            if (target == null)
+            if (target == null || Core.Me.CurrentTarget == null)
```
This seems weird to me, since the code is explicitly trying to find an AOE target, I don't know if we should return false if our current target is null? I guess it is weird to AOE when we don't have anything targeted. 

If you think this is right I'm good with it, but it seems odd to me in any case.

wildchill commented on Magitek/Logic/Summoner/Aoe.cs:
```diff
@@ -91,7 +91,8 @@ public static async Task<bool> CrimsonCyclone()
 
-            if (target == null)
+            //Add another target check so it works if SmartAoe is not enabled
+            if (target == null || Core.Me.CurrentTarget == null)
```
This seems weird to me, since the code is explicitly trying to find an AOE target, I don't know if we should return false if our current target is null? I guess it is weird to AOE when we don't have anything targeted. 

If you think this is right I'm good with it, but it seems odd to me in any case.

wildchill commented on Magitek/Logic/Summoner/Aoe.cs:
```diff
@@ -131,7 +132,7 @@ public static async Task<bool> MountainBuster()
             var target = Combat.SmartAoeTarget(Spells.MountainBuster, SummonerSettings.Instance.SmartAoe);
 
-            if (target == null)
+            if (target == null || Core.Me.CurrentTarget == null)
```
This seems weird to me, since the code is explicitly trying to find an AOE target, I don't know if we should return false if our current target is null? I guess it is weird to AOE when we don't have anything targeted. 

If you think this is right I'm good with it, but it seems odd to me in any case.

wildchill commented on Magitek/Logic/Summoner/Aoe.cs:
```diff
@@ -153,7 +154,7 @@ public static async Task<bool> Slipstream()
             var target = Combat.SmartAoeTarget(Spells.Slipstream, SummonerSettings.Instance.SmartAoe);
 
-            if (target == null)
+            if (target == null || Core.Me.CurrentTarget == null)
```
This seems weird to me, since the code is explicitly trying to find an AOE target, I don't know if we should return false if our current target is null? I guess it is weird to AOE when we don't have anything targeted. 

If you think this is right I'm good with it, but it seems odd to me in any case.

wildchill commented on Magitek/Logic/Summoner/Aoe.cs:
```diff
@@ -180,7 +181,7 @@ public static async Task<bool> EnergySiphon()
             var target = Combat.SmartAoeTarget(Spells.EnergySiphon, SummonerSettings.Instance.SmartAoe);
 
-            if (target == null)
+            if (target == null || Core.Me.CurrentTarget == null)
```
This seems weird to me, since the code is explicitly trying to find an AOE target, I don't know if we should return false if our current target is null? I guess it is weird to AOE when we don't have anything targeted. 

If you think this is right I'm good with it, but it seems odd to me in any case.

wildchill commented on Magitek/Logic/Summoner/Aoe.cs:
```diff
@@ -203,15 +204,15 @@ public static async Task<bool> Outburst()
                 target = Combat.SmartAoeTarget(Spells.BrandofPurgatory, SummonerSettings.Instance.SmartAoe);
 
-                if (target == null)
+                if (target == null || Core.Me.CurrentTarget == null)
```
This seems weird to me, since the code is explicitly trying to find an AOE target, I don't know if we should return false if our current target is null? I guess it is weird to AOE when we don't have anything targeted. 

If you think this is right I'm good with it, but it seems odd to me in any case.

wildchill commented on Magitek/Logic/Summoner/Aoe.cs:
```diff
@@ -203,15 +204,15 @@ public static async Task<bool> Outburst()
             target = Combat.SmartAoeTarget(Spells.PreciousBrilliance, SummonerSettings.Instance.SmartAoe);
             
-            if (target == null)
+            if (target == null || Core.Me.CurrentTarget == null)
```
This seems weird to me, since the code is explicitly trying to find an AOE target, I don't know if we should return false if our current target is null? I guess it is weird to AOE when we don't have anything targeted. 

If you think this is right I'm good with it, but it seems odd to me in any case.

wildchill commented on Magitek/Logic/Summoner/Aoe.cs:
```diff
@@ -282,7 +283,7 @@ public static async Task<bool> Painflare()
             var target = Combat.SmartAoeTarget(Spells.Painflare, SummonerSettings.Instance.SmartAoe);
 
-            if (target == null)
+            if (target == null || Core.Me.CurrentTarget == null)
```
This seems weird to me, since the code is explicitly trying to find an AOE target, I don't know if we should return false if our current target is null? I guess it is weird to AOE when we don't have anything targeted. 

If you think this is right I'm good with it, but it seems odd to me in any case.

wildchill commented on Magitek/Logic/Summoner/Aoe.cs:
```diff
@@ -314,7 +315,7 @@ public static async Task<bool> Ruin4()
             var target = Combat.SmartAoeTarget(Spells.Ruin4, SummonerSettings.Instance.SmartAoe);
 
-            if (target == null)
+            if (target == null || Core.Me.CurrentTarget == null)
```
This seems weird to me, since the code is explicitly trying to find an AOE target, I don't know if we should return false if our current target is null? I guess it is weird to AOE when we don't have anything targeted. 

If you think this is right I'm good with it, but it seems odd to me in any case.

wildchill reviewed (changes requested):
Made some notes for review
